examples/tools/meson/create_your_first_package.rst

```diff
@@ -0,0 +1,162 @@
+
+- **conanfile.py**: On the root folder, there is a *conanfile.py* which is the main recipe
+  file, responsible for defining how the package is built and consumed.
+- **meson.build**: A Meson build script. This script doesn't need to contain anything Conan-specific,
+  it is completely agnostic of Conan, because the integration is transparent.
+- **src** folder: the folder that contains the simple C++ "hello" library.
+- **test_package** folder: contains an *example* application that will require
+  and link with the created package. In this case the ``test_package`` also contains a
+  ``meson.build``, but it is possible to have the ``test_package`` using
+  other build system as CMake if desired. It is not mandatory that the test_package is using
+  the same build system as the package.
+
+Let's have a look at the package recipe *conanfile.py* (only the relevant new parts):
+
+.. code-block:: python
+
+    exports_sources = "meson.build", "src/*"
+
+    def layout(self):
+        basic_layout(self)
+
+    def generate(self):
+        tc = MesonToolchain(self)
+        tc.generate()
+
+    def build(self):
+        meson = Meson(self)
+        meson.configure()
+        meson.build()
+
+    def package(self):
+        meson = Meson(self)
+        meson.install()
+
+Let's explain the different sections of the recipe briefly:
+
+- The ``layout()`` defines a ``basic_layout()``, this is less flexible than a CMake one, so it
+  doesn't allow any parametrization.
+- The ``generate()`` method calls ``MesonToolchain`` that can generate ``conan_meson_native.ini``
+  and ``conan_meson_cross.ini`` Meson toolchain files for cross builds. If the project had dependencies
+  with Conan ``requires``, it should add ``PkgConfigDeps`` too
+- The ``build()`` method uses the ``Meson()`` helper to drive the build
+- The ``package()`` method uses the ``Meson`` install functionality to define and copy to the package 
+  folder the final artifacts.
+
+
+The **test_package** folder also contains a ``meson.build`` file that declares a dependency to
+the tested package, and links an application, to verify the package was correctly created and contains
+that library:
+
+.. code-block:: 
+  :caption: test_package/meson.build
+
+  project('Testhello', 'cpp')
+  hello = dependency('hello', version : '>=0.1')
+  executable('example', 'src/example.cpp', dependencies: hello)
+
+
+Note the ``test_package/conanfile.py`` contains also a ``generators = "PkgConfigDeps", "MesonToolchain"``,
+because the ``test_package`` has the "hello" package as dependency, and ``PkgConfigDeps`` is necessary to
+locate it.
+
+.. note:: 
+
+  This example assumes Meson, Ninja and PkgConfig are installed in the system, which might not always be the case.
+  If they are not, you can create a profile ``myprofile`` with:
+
+  .. code-block::
+
+    include(default)
+    
+    [tool_requires]
+    meson/[*]
+    pkgconf/[*]
+
+  and use it in ``conan create . -pr=myprofile``
```

Comment:
I agree, as this is for the beginner, maybe we can add a reference to the profiles section in case that addition to the profile sounds weird for them:

```suggestion
We added `Meson` and `pkgconf` as :ref:`tool requirements to the profile <reference_config_files_profiles_tool_requires>`. By executing ``conan create . -pr=myprofile``, those tools will be installed and made available during the package's build process.
```